src/validation-engine.js

```diff
@@ -42,6 +43,7 @@ class ValidationEngine {
     this.validationError = null
     try {
       this.initReport()
+      this.checkedNodes.clear()
```

Comment:
Clone is creating a new instance so this would no longer be necessary, right?
```suggestion
```

Reply:
Unless `validateAll` is called multiple time...

Reply:
Yes, I think calling it multiple times should be supported. Still I moved creation/clearing to initReport()